internal/application/utils/utils.go

```diff
@@ -30,3 +33,15 @@ func IsAllowedGrantType(oauthApplication *model.OAuthApplication, grantType stri
 	}
 	return false
 }
+
+// IsValidRedirectURI checks if the redirect URI is valid for the given OAuth application.
+func IsValidRedirectURI(oauthApplication *model.OAuthApplication, redirectURI string) bool {
+
+	// Check if the redirect URI is valid for the given OAuth application.
+	for _, allowedRedirectURI := range oauthApplication.RedirectURIs {
+		if redirectURI == allowedRedirectURI {
+			return true
+		}
+	}
+	return false
+}
```

Comment:
Can't we use this util fn as a fn in the OAuthApplication struct? it will be more clearer and maintainable

Reply:
Addressed with 69b1982c7d3bf4a6e2a9233f5ac199384c045518